tests/e2e/validatecluster/validatecluster_test.go

```diff
@@ -12,61 +13,279 @@ import (
-			fmt.Println("Agent Nodes:", agentNodeNames)
-			kubeConfigFile, err = e2e.GenKubeConfigFile(serverNodeNames[0])
+			fmt.Println("OS:", *nodeOS)
+			fmt.Println("Server Nodes:", serverNodenames)
+			fmt.Println("Agent Nodes:", agentNodenames)
+			kubeConfigFile, err = e2e.GenKubeConfigFile(serverNodenames[0])
 			Expect(err).NotTo(HaveOccurred())
 		})
-		It("Verify Node and Pod Status", func() {
+
+		It("Checks Node and Pod Status", func() {
 			fmt.Printf("\nFetching node status\n")
 			Eventually(func(g Gomega) {
-				nodes, err := e2e.ParseNodes(kubeConfigFile, true)
+				nodes, err := e2e.ParseNodes(kubeConfigFile, false)
 				g.Expect(err).NotTo(HaveOccurred())
 				for _, node := range nodes {
-					g.Expect(node.Status).Should(Equal("Ready"), func() string { return node.Name })
+					g.Expect(node.Status).Should(Equal("Ready"))
 				}
 			}, "420s", "5s").Should(Succeed())
+			_, _ = e2e.ParseNodes(kubeConfigFile, true)
 
 			fmt.Printf("\nFetching Pods status\n")
 			Eventually(func(g Gomega) {
-				pods, err := e2e.ParsePods(kubeConfigFile, true)
+				pods, err := e2e.ParsePods(kubeConfigFile, false)
 				g.Expect(err).NotTo(HaveOccurred())
 				for _, pod := range pods {
 					if strings.Contains(pod.Name, "helm-install") {
-						g.Expect(pod.Status).Should(Equal("Completed"), func() string { return pod.Name })
+						g.Expect(pod.Status).Should(Equal("Completed"), pod.Name)
 					} else {
-						g.Expect(pod.Status).Should(Equal("Running"), func() string { return pod.Name })
+						g.Expect(pod.Status).Should(Equal("Running"), pod.Name)
 					}
 				}
 			}, "420s", "5s").Should(Succeed())
+			_, _ = e2e.ParsePods(kubeConfigFile, true)
+		})
+
+		It("Verifies ClusterIP Service", func() {
+			_, err := e2e.DeployWorkload("clusterip.yaml", kubeConfigFile, false)
+			if err != nil {
+				fmt.Println("Cluster IP manifest not deployed", err)
+				Expect(err).NotTo(HaveOccurred())
+			}
+			Eventually(func(g Gomega) {
+				cmd := "kubectl get pods -o=name -l k8s-app=nginx-app-clusterip --field-selector=status.phase=Running --kubeconfig=" + kubeConfigFile
+				res, err := e2e.RunCommand(cmd)
+				Expect(err).NotTo(HaveOccurred())
+				g.Expect(res).Should((ContainSubstring("test-clusterip")))
+			}, "240s", "5s").Should(Succeed())
+
+			clusterip, _ := e2e.FetchClusterIP(kubeConfigFile, "nginx-clusterip-svc")
+			cmd := "\"curl -L --insecure http://" + clusterip + "/name.html\""
+			fmt.Println(cmd)
+			for _, nodeName := range serverNodenames {
+				Eventually(func(g Gomega) {
+					res, err := e2e.RunCmdOnNode(cmd, nodeName)
+					g.Expect(err).NotTo(HaveOccurred())
+					fmt.Println(res)
+					Expect(res).Should(ContainSubstring("test-clusterip"))
+				}, "120s", "10s").Should(Succeed())
+			}
+		})
+
+		It("Verifies NodePort Service", func() {
+			_, err := e2e.DeployWorkload("nodeport.yaml", kubeConfigFile, false)
+			if err != nil {
+				fmt.Println("NodePort manifest not deployed", err)
+				Expect(err).NotTo(HaveOccurred())
+			}
```

Comment:
Can be combined into one line
```suggestion
			Expect(err).NotTo(HaveOccurred(), "NodePort manifest not deployed")
```